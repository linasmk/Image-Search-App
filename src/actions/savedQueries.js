/* ========= Dependencies ============= */
import { v4 as uuidv4 } from "uuid";
/* ========= Imports ============= */
import { ADD_SAVED_QUERY, REMOVE_SAVED_QUERY } from "@consts/constants";
/* =================================================
=========== ACTION CREATORS: SAVED QUERIES ========
================================================= */

export const addSavedQuery = ({ name = "" } = {}) => ({
  type: ADD_SAVED_QUERY,
  savedQuery: {
    id: uuidv4(),
    name
  }
});

export const removeSavedQuery = ({ id } = {}) => ({
  type: REMOVE_SAVED_QUERY,
  id
});
