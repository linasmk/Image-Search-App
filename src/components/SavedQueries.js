/* ========= App Dependencies ============= */
import React, { useState } from "react";
import PropTypes from "prop-types";
/* =========== Redux ============== */

/* ========= Components =============== */
import { SavedQueryItem } from "@cont/SavedQueryItem";
/* ========= Code ============= */
export const SavedQueries = ({ savedQueries, onClick }) => {
  const [sliceSavedQueries, setSliceSavedQueries] = useState(10);
  const [queriesListOpen, setQueriesListOpen] = useState(false);

  const toggleQueriesList = () => {
    if (queriesListOpen) {
      setSliceSavedQueries(10);
    } else {
      setSliceSavedQueries(savedQueries.length);
    }
    setQueriesListOpen(!queriesListOpen);
  };
  return (
    <section className="saved-queries">
      <div className="saved-queries__wrapper">
        {savedQueries
          .map((savedQuery) => (
            <SavedQueryItem
              key={savedQuery.id}
              id={savedQuery.id}
              name={savedQuery.name}
              onClick={onClick}
            />
          ))
          .slice(0, sliceSavedQueries)}
        <div className="saved-queries__btn-wrapper">
          <button
            className="saved-queries__btn--show"
            type="button"
            style={
              savedQueries.length > 10
                ? { display: "block" }
                : { display: "none" }
            }
            onClick={toggleQueriesList}>
            {queriesListOpen ? "Show less" : "Show more"}
          </button>
        </div>
      </div>
    </section>
  );
};
SavedQueries.defaultProps = {
  onClick: undefined
};
SavedQueries.propTypes = {
  savedQueries: PropTypes.arrayOf(PropTypes.object).isRequired,
  onClick: PropTypes.func
};
export default SavedQueries;
