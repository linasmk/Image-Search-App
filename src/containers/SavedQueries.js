/* ===== Redux ===== */
import { connect } from "react-redux";
import latestQueryFirst from "@selectors/latestQueryFirst";

/* ====== Components ===== */
import SavedQueriesComponent from "@comp/SavedQueries";

/* ========= Code ========= */
export const SavedQueries = connect((state) => ({
  savedQueries: latestQueryFirst(state.savedQueries)
}))(SavedQueriesComponent);
