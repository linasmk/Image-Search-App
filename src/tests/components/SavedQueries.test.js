/* ========= App Dependencies ============= */
import React from "react";
import { shallow } from "enzyme";
/* ========= Components ============= */
import { SavedQueries } from "@comp/SavedQueries";
import savedQueries from "@tests/fixtures/queries";
/* ========= Code ============= */

it("renders SavedQueryItem component with SavedQueries array correctly", () => {
  const wrapper = shallow(<SavedQueries savedQueries={savedQueries} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders SavedQueryItem component with empty SavedQueries array correctly", () => {
  const wrapper = shallow(<SavedQueries savedQueries={[]} />);
  expect(wrapper).toMatchSnapshot();
});
