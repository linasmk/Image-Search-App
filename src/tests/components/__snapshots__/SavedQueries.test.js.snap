// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SavedQueryItem component with SavedQueries array correctly 1`] = `
<section
  className="saved-queries"
>
  <div
    className="saved-queries__wrapper"
  >
    <Connect(SavedQueryItem)
      id="1"
      key="1"
      name="dog"
    />
    <Connect(SavedQueryItem)
      id="2"
      key="2"
      name="lemon"
    />
    <Connect(SavedQueryItem)
      id="3"
      key="3"
      name="notes"
    />
    <div
      className="saved-queries__btn-wrapper"
    >
      <button
        className="saved-queries__btn--show"
        onClick={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="button"
      >
        Show more
      </button>
    </div>
  </div>
</section>
`;

exports[`renders SavedQueryItem component with empty SavedQueries array correctly 1`] = `
<section
  className="saved-queries"
>
  <div
    className="saved-queries__wrapper"
  >
    <div
      className="saved-queries__btn-wrapper"
    >
      <button
        className="saved-queries__btn--show"
        onClick={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="button"
      >
        Show more
      </button>
    </div>
  </div>
</section>
`;
